src/Okta.Sdk.Abstractions/DefaultDataStore.cs

```diff
@@ -160,7 +166,18 @@ private void EnsureResponseSuccess(HttpResponse<string> response)
                 throw new InvalidOperationException($"An error occurred deserializing the error body for response code {response.StatusCode}. See the inner exception for details.", ex);
             }
 
-            throw new OktaApiException(response.StatusCode, _resourceFactory.CreateNew<ApiError>(errorData));
+            var contentTypeHeader = response.Headers.FirstOrDefault(x => string.Equals(x.Key, "Content-Type", StringComparison.OrdinalIgnoreCase));
+
+            // If error format is ION
+            if (!contentTypeHeader.Equals(default(KeyValuePair<string, string>))
+                && contentTypeHeader.Value.FirstOrDefault().Contains("application/ion+json"))
+            {
+                throw new OktaIonApiException(response.StatusCode, _resourceFactory.CreateNew<IonApiError>(errorData));
+            }
+            else
+            {
+                throw new OktaApiException(response.StatusCode, _resourceFactory.CreateNew<ApiError>(errorData));
+            }
```

Comment:
The work done here would be better served by a "Content-Type" to exception type dictionary; this would make adding future exception types fairly trivial.  Maybe not serious enough to warrant a change now if time is of concern.